### PR TITLE
Update Payments Page to show wallet balance, empty state, and more!

### PR DESCRIPTION
**Pull request by stitesExpensify**
### Details
This PR improves the Payments page and brings it more in line with the original design.
Specifically:
1. Adds the wallet balance
2. Removes the wallet as a payment method to fund the wallet
3. Adds the "Payment Methods" title
4. Adds the empty state for the payment methods list
5. Adds a loading indicator for when we don't have the wallet balance

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/169463

### Tests / QA
1. Create a new account
2. Go to Settings -> Payments
3. Make sure you see this empty state

4. Log in with an account that has bank accounts/cards
5. Make sure you see this state

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

File: src/components/CurrentWalletBalance.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            style={[styles.textXXXLarge, styles.pv5, styles.alignSelfCenter]}
+        >
+            {`$${formattedBalance}`}
+        </Text>
+    );
+};
+
+CurrentWalletBalance.propTypes = propTypes;
+CurrentWalletBalance.defaultProps = defaultProps;
+CurrentWalletBalance.displayName = 'ConfirmModal';
```
Comment (jasperhuangg) [inline src/components/CurrentWalletBalance.js:50]:
```suggestion
CurrentWalletBalance.displayName = 'CurrentWalletBalance';
```

File: src/libs/actions/PaymentMethods.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ function getPaymentMethods() {
     })
         .then((response) => {
             Onyx.multiSet({
-                [ONYXKEYS.USER_WALLET]: lodashGet(response, 'userWallet', null),
+                [ONYXKEYS.USER_WALLET]: lodashGet(response, 'userWallet', {}),
```
Comment (jasperhuangg) [inline src/libs/actions/PaymentMethods.js:17]: Just curious, any reason you decided to default this from `null` to `{}`? AFAIK it doesn't really change anything if we left it as `null`.

**Review by jasperhuangg — CHANGES_REQUESTED** [review]
Besides a super minor change and a quick question, the code looks great and tests well, good stuff :)

I'm wondering whether we should display a loader for the entire modal or disable the "Add Payment Method" button while we're waiting to grab the user's payment methods. Otherwise, it can cause the (kinda ugly/confusing) scenario depicted here where the user opens the dropdown but then the anchor for the dropdown moves:

https://user-images.githubusercontent.com/31285285/124702796-a6c96d00-df23-11eb-80b4-a17940a6c67e.mp4

File: src/components/CurrentWalletBalance.js (hunk 1 above)
Comment (stitesExpensify) [reply src/components/CurrentWalletBalance.js:50]: 🍝 ty

File: src/libs/actions/PaymentMethods.js (hunk 2 above)
Comment (stitesExpensify) [reply src/libs/actions/PaymentMethods.js:17]: I'm checking for `_.isEmpty` in `CurrentWalletBalance` so it just made more sense to me /shrug

**Comment by jasperhuangg** [comment]
@stitesExpensify Were you able to look into that issue I mentioned above?

**Comment by stitesExpensify** [comment]
Hey there! Sorry for not updating you, but I'm adding a loading circle as we speak :)

**Comment by stitesExpensify** [comment]
Updated!

**Comment by jasperhuangg** [comment]
Hey @stitesExpensify, no worries! I just tested out your solution and it doesn't really seem like it fixes the problem I brought up earlier? I think we should just disable that button until everything is loaded:


https://user-images.githubusercontent.com/31285285/125147996-d2d23180-e161-11eb-9fda-f756f9079b99.mp4

@Expensify/design What are your thoughts?

**Comment by shawnborton** [comment]
Ah, I see. That makes sense to me. I think we could just "disable" the Add Payment row and make it like 50% opacity like we do with disabled buttons. Thoughts on something like that?

**Comment by jasperhuangg** [comment]
> Ah, I see. That makes sense to me. I think we could just "disable" the Add Payment row and make it like 50% opacity like we do with disabled buttons. Thoughts on something like that?

Yup! I totally agree, that was my original recommendation. Should be a one-liner too.

cc @stitesExpensify

**Comment by stitesExpensify** [comment]
Updated to disable the input when loading :)

**Comment by jasperhuangg** [comment]
Hey @stitesExpensify sorry for the delay!

**Review by jasperhuangg — COMMENTED** [review]
@stitesExpensify Looks good! Let's also get rid of the smaller loading spinner you added since it seems kinda redundant now?

**Comment by stitesExpensify** [comment]
> since it seems kinda redundant now

I disagree, I think that showing when the methods are loading is still a useful UI element. A disabled button doesn't show that things are loading in and of itself. I think the loader should stay

**Comment by jasperhuangg** [comment]
@stitesExpensify Sorry for the late response; I've been traveling back to LA. I know this PR is really simple and it's definitely been ongoing for a long time, so sorry for that as well!

I still don't really understand why we need both loaders still. Correct me if I'm wrong, but it looks like both loaders should disappear simultaneously once [`getPaymentMethods`](https://github.com/Expensify/App/pull/3890/files#diff-9f03259c1365af851e911d6200cd07981e2ff68fff1b09f9105f4262dcab7943R12-R22) returns and updates all the Onyx keys that both loaders depend on in `multiSet`.

Also, from a pure UX standpoint, I think we should get someone from @Expensify/design to help us decide on how it looks.

Here's the screen recording for reference:
https://user-images.githubusercontent.com/31285285/125147996-d2d23180-e161-11eb-9fda-f756f9079b99.mp4

**Comment by shawnborton** [comment]
Candidly I'm a little lost on what is going on here, but looking at the video that Jasper just posted, I agree that the two loading spinners feel strange. Could we just have one that loads everything?

**Comment by stitesExpensify** [comment]
🤦 You're right. Totally my bad, sorry.  I forgot that the wallet balance also has a loader. I think we should just keep the balance loader and get rid of the smaller one. Sound good @shawnborton ?

**Comment by shawnborton** [comment]
Sounds good!

**Comment by stitesExpensify** [comment]
Sorry again about that @jasperhuangg. Updated!

**Review by jasperhuangg — APPROVED** [review]
Looks great! Thanks for your patience with the review :)

**Pull request merged**